app/api/chat/route.ts

```diff
@@ -44,9 +45,21 @@ export async function POST(request: Request) {
       .execute();
 
     if (!contact.length) {
+      const contactData = { email };
+
+      if (!shouldCreateContact(contactData)) {
+        return NextResponse.json(
+          {
+            error:
+              "Cannot create chat session: no identifying information provided",
+          },
+          { status: 400 }
+        );
+      }
+
       const newContact = await db
         .insert(ContactsTable)
-        .values({ email })
+        .values(contactData)
         .returning({ id: ContactsTable.id });
```

Comment:
centralize this should create and create logic into a util in `/server` that raises a custom exception if `shouldCreateContact` is false